lib/Pipeline/Step.cpp

```diff
@@ -106,6 +101,23 @@ Step::cloneAndRun(Context &Ctx, ContainerSet &&Input, llvm::raw_ostream *OS) {
   return Containers.cloneFiltered(InputEnumeration);
 }
 
+void Step::runAnalysis(llvm::StringRef AnalysisName,
+                       Context &Ctx,
+                       const ContainerToTargetsMap &Targets,
+                       llvm::raw_ostream *OS) {
+  ContainerToTargetsMap Map = Containers.enumerate();
+  revng_check(Map.contains(Targets),
+              "An analysis was requested, but not all targets are aviable");
```

Comment:
Let's use `revng_assert`, I'd keep `revng_check` only for stuff we deem to be security/safety-sensitive.
If we don't have a clear policy future refactorings will make us cry.

On the other hand, if you think this might be due to a user mistake, we need to return an error, not abort.

Reply:
yeah, the issue here is that we need to do that pervasive change were we allow to return errors in run and analyze

Reply:
Hopefully `Pipe::run` should not fail, we discussed about checking prerequisites, but that's a different story.
Here, either we return a graceful error or we should run a pipeline to ensure those targets are produced, which is how I thought this would work, or am I missing some reason for which this not viable?

Reply:
ah right yeah, if this check is triggered it is a pipe implementation issue, can be a assert